jwst/associations/lib/rules_level3.py

```diff
@@ -667,6 +667,25 @@ def __init__(self, *args, **kwargs):
                 ],
                 reduce=Constraint.notany
             ),
+            # Don't allow NIRCam engineering mode
+            # with PUPIL='CLEAR' in tso3
+            Constraint(
+                [
+                    Constraint([
+                        DMSAttrConstraint(
+                            name='restricted_grism',
```

Comment:
Because the containing Asn Class already has another constraint involving exp_type, I wasn't sure if it was OK to reuse the same `name` value, so I followed the lead of another rule and made up a different name, `restricted_grism`.